evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
-        return vm_class
+        for n in reversed(cls.vms_by_range.keys()):
```

Comment:
Maybe I'm mistaken but I think a binary search here wouldn't provide any significant improvements given that most of the time this will be called for the latest block number, in which case we'd almost invariably pick the last item from vms_by_range, no?  But even if that's not the case, I also wonder if vms_by_range will ever have enough items to warrant a binary search?